record-and-playback/presentation/scripts/process/presentation.rb

```diff
@@ -48,105 +49,95 @@
 log_dir = props['log_dir']
 
 target_dir = "#{recording_dir}/process/presentation/#{meeting_id}"
-if not FileTest.directory?(target_dir)
+unless FileTest.directory?(target_dir)
   FileUtils.mkdir_p "#{log_dir}/presentation"
-  logger = Logger.new("#{log_dir}/presentation/process-#{meeting_id}.log", 'daily' )
+  logger = Logger.new("#{log_dir}/presentation/process-#{meeting_id}.log", 'daily')
   BigBlueButton.logger = logger
-  BigBlueButton.logger.info("Processing script presentation.rb")
+  BigBlueButton.logger.info('Processing script presentation.rb')
   FileUtils.mkdir_p target_dir
 
   begin
-    # Create a copy of the raw archives
-    temp_dir = "#{target_dir}/temp"
-    FileUtils.mkdir_p temp_dir
-    FileUtils.cp_r(raw_archive_dir, temp_dir)
-
     # Create initial metadata.xml
-    b = Builder::XmlMarkup.new(:indent => 2)
-    metaxml = b.recording {
+    b = Builder::XmlMarkup.new(indent: 2)
+    metaxml = b.recording do
       b.id(meeting_id)
-      b.state("processing")
+      b.state('processing')
       b.published(false)
       b.start_time
       b.end_time
       b.participants
       b.playback
       b.meta
-    }
-    metadata_xml = File.new("#{target_dir}/metadata.xml","w")
+    end
+    metadata_xml = File.new("#{target_dir}/metadata.xml", 'w')
     metadata_xml.write(metaxml)
     metadata_xml.close
-    BigBlueButton.logger.info("Created inital metadata.xml")
+    BigBlueButton.logger.info('Created inital metadata.xml')
 
-    BigBlueButton::AudioProcessor.process("#{temp_dir}/#{meeting_id}", "#{target_dir}/audio")
-    events_xml = "#{temp_dir}/#{meeting_id}/events.xml"
+    BigBlueButton::AudioProcessor.process(raw_archive_dir, "#{target_dir}/audio")
+    events_xml = "#{raw_archive_dir}/events.xml"
+
+    # TODO: Don't copy events.xml to target directory
     FileUtils.cp(events_xml, target_dir)
 
-    presentation_dir = "#{temp_dir}/#{meeting_id}/presentation"
+    presentation_dir = "#{raw_archive_dir}/presentation"
     presentations = BigBlueButton::Presentation.get_presentations(events_xml)
 
     processed_pres_dir = "#{target_dir}/presentation"
     FileUtils.mkdir_p processed_pres_dir
 
     # Get the real-time start and end timestamp
-    @doc = Nokogiri::XML(File.read("#{target_dir}/events.xml"))
+    @doc = Nokogiri::XML(File.read("#{target_dir}/events.xml")) 
 
-    meeting_start = @doc.xpath("//event")[0][:timestamp]
-    meeting_end = @doc.xpath("//event").last()[:timestamp]
+    event_timestamps = @doc.xpath('recording/event/@timestamp')
```

Comment:
This is better than the previous code, but best would be if you switch it to use the `first_event_timestamp` and `last_event_timestamps` provided in the shared code: https://github.com/bigbluebutton/bigbluebutton/blob/develop/record-and-playback/core/lib/recordandplayback/generators/events.rb#L80-L90

Reply:
Fixed.